Iss #254 adding additional fields to lidar config

kersting opened the pull request:
Added new fields to lidar_config table

1. logger_stated_device_datum_plane_height: Height of the datum plane as programmed into the logger
2. logger_stated_device_orientation_deg: Device orientation in degrees as programmed into the logger

Updated the following files

- iea43_wra_data_model.schema
- iea43_wra_dat_model_postgresssql.sql
- CHANGELOG.md
- floating_lidar_demo_iea43_wra_data_model.json

kersting committed:
Issue_254

Adding logger_stated_device_datum_plane_height and logger_stated_device_orientation_deg to lidar_config table

kersting committed:
Update floating_lidar_demo_iea43_wra_data_model.json

kersting committed:
Update iea43_wra_data_model.schema.json

Fixed tabs and spaces.

kersting committed:
Update iea43_wra_data_model_postgresql.sql

Fixed tabs and spaces

kersting committed:
Update CHANGELOG.md

Fixed problems with spaces and tabs.

kersting committed:
Update floating_lidar_demo_iea43_wra_data_model.json

Fixing spaces and tabs.

abohara commented on schema/iea43_wra_data_model.schema.json:
@kersting Should the type be a 'number' ?

abohara commented on tools/iea43_wra_data_model_postgresql.sql:
@kersting Shouldn't the type for both be of type number ( e.g. decimal ) instead of boolean

stephenholleran commented on CHANGELOG.md:
```diff
@@ -27,6 +30,9 @@ Additional labels for pre-release and build metadata are available as extensions
    1. `adcp` (Issue [#226](https://github.com/IEA-Task-43/digital_wra_data_standard/issues/226))
    1. `virtual_met_mast` (Issue [#214](https://github.com/IEA-Task-43/digital_wra_data_standard/issues/214))
    1. `reanalysis` (Issue [#214](https://github.com/IEA-Task-43/digital_wra_data_standard/issues/214))
+1. To `lidar_config` enum add:
```
These notes should go under line 12 as we have released version 1.3 already.

stephenholleran commented on demo_data/floating_lidar_demo_iea43_wra_data_model.json:
```diff
@@ -54,6 +54,16 @@
         "date_to": null,
         "update_at": "2022-03-31T12:13:27"
+       },{
+	"logger_stated_device_datum_plane_height": 1.5,
```
Would you not just add `logger_stated_device_datum_plane_height` to the existing config? The way you have it you have two lidar configs for the same point in time. One for a flow correction and the other for this.

kersting replied:
Problem fixed. Thanks for pointing out.

stephenholleran commented on schema/iea43_wra_data_model.schema.json:
```diff
@@ -992,6 +992,28 @@
                       },
+                      "logger_stated_device_datum_plane_height": {
+                        "type": [
+                          "boolean",
```
Shouldn't this be a "number" not "boolean"?
Same for the device orientation.

kersting replied:
Yes, I should stop working on Sundays... I'll get this fixed. Thanks for catching all the errors.

stephenholleran commented on schema/iea43_wra_data_model.schema.json:
```diff
@@ -992,6 +992,28 @@
                           "false (for no flow corrections applied)"
                         ]
                       },
+                      "logger_stated_device_datum_plane_height": {
```
Forgot the trailing `_m`.

This is what is in the Vertical Profiler Properties

```JSON
"device_datum_plane_height_m": {
                  "type": [
                    "number",
                    "null"
                  ],
"title": "Device Datum Plane Height [m]",
"description": "The datum plane height of the remote sensing device where the datum feature is defined here: http://data.windenergy.dtu.dk/ontologies/view/ontolidar/en/page/DatumFeature . For lidars the datum feature is usually the window and for sodars it is usually the base of the device. These datum plane heights are also usually referred to as above ground level however it may be above sea level or above a platform level.",
```

kersting committed:
Update iea43_wra_data_model_postgresql.sql

stephenholleran commented:
Hey @kersting,

Thanks for having a go with this. I made just some minor enough comments.

This PR however is merging directly into `master`. Could you change it to merge into the `dev` branch?

Cheers,

kersting committed:
Update iea43_wra_data_model.schema.json

kersting committed:
Update iea43_wra_data_model.schema.json

kersting committed:
Update floating_lidar_demo_iea43_wra_data_model.json

kersting committed:
Update iea43_wra_data_model_postgresql.sql

kersting committed:
Update iea43_wra_data_model_postgresql.sql

kersting committed:
Update CHANGELOG.md

kersting committed:
Update CHANGELOG.md

kersting commented:
@stephenholleran thank you for the review. I did my pull request on Sunday, it clearly was a mistake. I think I fixed everything as I have also moved the request to dev. Could you please do a last sanity check? Feel free to merge into dev if everything is good or you can just ping me here.

stephenholleran committed:
iss #254 update descriptions

stephenholleran commented:
#254 
Hi @kersting,

Thanks for the changes, it looks way better now.

I made some edits to the descriptions. Make sure you agree with them.

What do you think of adding in examples to the main demo file? https://github.com/IEA-Task-43/digital_wra_data_standard/blob/master/demo_data/iea43_wra_data_model.json

And while I was reviewing it I realised that the device datum plane height programmed into the device could be for a lidar or a sodar. However we have applied the logger_stated into the 'lidar_config' table. We might want to think about renaming this 'lidar_config' table to something else but that would be a breaking change. Though this is a bit confusing for someone setting up a sodar I think we can let this slide and follow up with a name change when we are doing other breaking changes.

@abohara it would be good to get your thoughts and review of this PR too?

Thanks,

kersting commented:
@stephenholleran thank you for your comments and for editing the descriptions. I'm fine with your edits.

As far as adding to the main demo file, I changed the floating lidar one instead as the main demo file is focused in met mast.

Nice catch regarding the lidar/sodar situation. I'm not opposed to rename from lidar_config to rsd_config. However, I feel that sodars are more of a legacy device at this point so I don't need to rush. On the other hand, I think we need to make the break change so maybe we may want to bundle with other items that will cause a break change as well.

stephenholleran commented:
> far as adding to the main demo file, I changed the floating lidar one instead as the main demo file is focused in met mast.

Ah yes, good point.

> Nice catch regarding the lidar/sodar situation. I'm not opposed to rename from lidar_config to rsd_config. However, I feel that sodars are more of a legacy device at this point so I don't need to rush. On the other hand, I think we need to make the break change so maybe we may want to bundle with other items that will cause a break change as well.

I think these changes can continue as they are but we should create another issue to change the name of the 'lidar_config' table to something like you suggest, 'rsd_config', and bundle it in with the other breaking changes we have lined up.

So, I am good with this PR. It would be good to get @abohara to cast his eye over it too before merging.

Thanks,

stephenholleran commented:
Hi @abohara,
Any chance you can take a look at this? This PR is updating the Schema so it would be great for you to review.
Cheers,

abohara commented on CHANGELOG.md:
```diff
@@ -10,8 +10,9 @@ Given a version number MAJOR.MINOR.PATCH, increment the:
 ## [Unreleased X.X.0-202X.XX]
-
-1. 
+1. To `lidar_config` enum add:
```
@kersting - Can we remove 'enum' and simply state " to 'lidar_config' add: " ?

abohara reviewed (approved):
Beside a minor comment, looks good @kersting . Thank you!

kersting committed:
Update CHANGELOG.md

kersting reviewed (commented):
@abohara thanks for catching that. I've removed the enum part.
